app/_hub/kong-inc/grpc-web/index.md

```diff
@@ -0,0 +1,158 @@
+---
+name: gRPC-Web
+publisher: Kong Inc.
+
+categories:
+  - transformations
+
+type: plugin
+
+desc: Allow browser clients to call gRPC services.
+description: |
+  A Kong plugin to allow access to a gRPC service via the gRPC-Web protocol.
+  Primarily, this means JS browser apps using the gRPC-Web library.
+  
+source_url: https://github.com/Kong/kong-plugin-grpc-web
+
+license_type: MIT
+
+kong_version_compatibility:
+  community_edition:
+    compatible:
+      - 2.0.x
+      - 1.5.x
+      - 1.4.x
+
+
+params:
+  name: grpc-web
+  route_id: true
+  protocols: ["http", "https"]
+  dbless_compatible: true
+
+  config:
+    - name: proto
+      required: false
+      default:
+      value_in_examples: path/to/hello.proto
```

Comment:
Storing filesystem path in the database is not a common practice in Kong (say, expect a few very specific case such as the File Log plugin). Instead we store those file's content directly inside the database (like certificates). This makes deployment and config management a lot easier.

(I know this is not the best place for feature related comments, but it seems that the other PR has been closed already because of the new repo).

Reply:
It can certainly be done, since the `protoc.lua` compiler takes a string as readily as a file; but the usual workflow on gRPC is to share the same .proto file between client and server, usually pulling from a common repository that defines the protocol for both (or a monorepo?  it's mostly google, after all).
Copying the content into the config (or database) would add extra work to keep client and server in sync across protocol evolution.